lm75bd/lm75bd.c

```
@@ -26,9 +26,35 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
-  return ERR_CODE_SUCCESS;
+    /* Implement this driver function */
+    if (temp == NULL) return ERR_CODE_INVALID_ARG;
+
+    error_code_t errCode;
+    uint8_t tempRegAddr = 0x00U;
+    uint8_t buff[2];
+
+    //Send address of temperature register to sensor
+    errCode = i2cSendTo(devAddr, &tempRegAddr, 1);
+
+    if(errCode != ERR_CODE_SUCCESS) return errCode;
```

If either I2C function fails, we don't want to just return the error code, we want to log it. There are some pre-defined macros you can use for this purpose; look for one that both logs and returns the error code.